### DISCUSSION TITLE
How to update Pinia state in response interceptor?

### DISCUSSION THREAD

**Discussion by hmingv** [nuxt/nuxt#16731] [Questions]
Hi everyone, I'm using JWT.

When the token expires, my API returns an Authorization in the headers.

It contains a new Token, and I need to use this new Token to replace the Token stored in pinia so that the API can be accessed normally next time.

I did and But it didn't work.

After the response interceptor is modified, it only takes effect temporarily, and the client does not change.

Excuse me, is there any way I can modify the value of Pinia in the response of the request?

![image](https://user-images.githubusercontent.com/57735185/192713295-709aacec-232f-4f06-b53b-11c07962b646.png)

![image](https://user-images.githubusercontent.com/57735185/192712046-31974087-a234-44c3-97f5-9a50bfaa62ea.png)

![image](https://user-images.githubusercontent.com/57735185/192712151-cf69c004-1827-4aa3-8298-ebb24a3d8fee.png)

**Comment by hmingv** [comment]
https://github.com/unjs/ohmyfetch/issues/134

**Comment by danielroe** [comment]
Would you create a reproduction so we can talk about some code?

**Reply by hmingv** [reply]
I just made one of this, but I'm not using pinia, I'm using useState, If they differ on this question, then I can revise it.


I used vitesse-nuxt3 as a template, I tried a few basic ones, and they didn't work well.

[reproduction](https://codesandbox.io/p/github/antfu/vitesse-nuxt3/csb-h68z9e/draft/delicate-fog?file=%2Fcomposables%2FuseCustomFetch.ts&selection=%5B%7B%22endColumn%22%3A26%2C%22endLineNumber%22%3A5%2C%22startColumn%22%3A26%2C%22startLineNumber%22%3A5%7D%5D&workspace=%257B%2522activeFileId%2522%253A%2522cl8lfvw8k03ju376hfjm4szke%2522%252C%2522openFiles%2522%253A%255B%2522%252Fcomposables%252Fcount.ts%2522%252C%2522%252Fcomposables%252FuseCustomFetch.ts%2522%252C%2522%252Fpages%252Findex.vue%2522%255D%252C%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522gitSidebarPanel%2522%253A%2522COMMIT%2522%252C%2522sidekickItems%2522%253A%255B%257B%2522type%2522%253A%2522PREVIEW%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522key%2522%253A%2522cl8lfjgaa00i3376hxdv43cfk%2522%252C%2522isMinimized%2522%253Afalse%252C%2522path%2522%253A%2522%252F%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522key%2522%253A%2522cl8lfi9zh004r376h7o5i1v8d%2522%252C%2522isMinimized%2522%253Afalse%257D%255D%257D)

**Reply by hmingv** [reply]
Thank you very much!
